spec/ruby/core/kernel/eval_spec.rb

```diff
@@ -141,6 +141,17 @@
     }
   end
 
+  ruby_version_is "3.2" do
+    it "includes path in syntax error" do
+      expected = 'speccing.rb'
+      -> {
+        eval('if true',TOPLEVEL_BINDING, expected)
+      }.should raise_error(SyntaxError) { |e|
+        e.path.should == expected
+      }
+    end
+  end
```

Comment:
I suppose we also need a case without `eval`. For instance we can `require` some fixture Ruby source file and check that `#path` returns a path to this file.

Reply:
**suggestion:** It seems the proper place for testing `SyntaxError#path` is `spec/ruby/core/exception` directory. There is no file for `SyntaxError` class yet so it could be created and contain something like this:

```ruby
require_relative '../../spec_helper'

describe 'SyntaxError#path' do
  ...
end
```

Reply:
Added!

Reply:
Could you move this spec there too?